src/plugins/yarn.js

```diff
@@ -24,19 +38,18 @@ export default class Yarn {
 
   get version (): string { return require('../../package.json')['cli-engine']['yarnDependency'] }
   get lockfile (): string { return path.join(this.config.cacheDir, 'yarn.lock') }
-  get bin (): ?string { return path.join(__dirname, '..', '..', 'yarn', `yarn-${this.version}.js`) }
+  get bin (): string { return path.join(__dirname, '..', '..', 'yarn', `yarn-${this.version}.js`) }
 
   async exec (args: string[] = [], options: {} = {}): Promise<string> {
     options = Object.assign({
       cwd: path.join(this.config.dataDir, 'plugins'),
-      stdio: this.config.debug ? 'inherit' : null
+      stdio: this.config.debug ? [0, 1, 2, 'ipc'] : [null, null, null, 'ipc']
```

Comment:
@dickeyxxx I was not sure how this would work, so I tested with a simple test program and when `this.config.debug` is true, then `stdout` does not go to `output.stdout` which I do not think that you want here.